hy/core/macros.hy

```diff
@@ -20,6 +20,15 @@
          ~@(interleave (repeat name) rest))
      ~name))
 
+
+(defmacro assoc [coll k1 v1 &rest other-kvs]
+  (if (odd? (len other-kvs))
+    (macro-error (last other-kvs) "`assoc` takes an odd number of arguments"))
+  (setv c (if other-kvs (gensym "c") coll))
+  `(setv ~@(+ (if other-kvs [c coll] []) #*
+    (genexpr [`(get ~c ~k) v] [[k v] (partition (+ (, k1 v1) other-kvs))]))))
```

Comment:
This is not legible, Kodi.
I cannot approve obfuscated code.

You don't seem to understand how to format Lisp, or why it is important to. Please read up on it, or let Emacs do it for you.